status/unstructured_adapter.go

```diff
@@ -0,0 +1,49 @@
+package status
+
+import (
+	"encoding/json"
+
+	"github.com/awslabs/operatorpkg/object"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// unstructuredAdapter is an adapter for the status.Object interface. unstructuredAdapter
+// makes the assumption that status conditions are found on status.conditions path.
+type unstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+func NewUnstructuredAdapter(obj client.Object) (*unstructuredAdapter, error) {
```

Comment:
Why are we returning an error from this adapter? Beyond the `lo.Must`, it's not possible for this constructor to throw an error